<?xml version="1.0" encoding="utf-8" ?>

<launch>

    <arg name="is_autostart" default="true"/>
    <arg name="frame_rate" default="10"/>

    <arg name="is_publish_bounding_box_image" default="true"/>
    <arg name="is_publish_bounding_box" default="true"/>

    <arg name="score_threshold" default="0.2"/> <!-- 0.2 -->
    <arg name="iou_threshold" default="0.7"/> <!-- 0.45 -->

    <arg name="config" default="$(find yolov5_ros)/configs/yolov5s.yaml"/>
    <arg name="weights_path" default="$(find yolov5_ros)/weights/best.pt"/>
    <!-- arg name="config" default="$(find yolov5_ros)/configs/yolov5m_irex2022_ito.yaml"/ -->
    <!-- arg name="weights_path" default="$(find yolov5_ros)/weights/yolov5m_irex2022_ito.pt"/ -->
    <arg name="device" default="cuda"/>

    <node pkg="yolov5_ros" type="yolov5_object_detector_node.py" name="yolov5_ros" output="screen">
        <param name="is_autostart" value="$(arg is_autostart)"/>
        <param name="frame_rate" value="$(arg frame_rate)"/>

        <param name="is_publish_bounding_box_image" value="$(arg is_publish_bounding_box_image)"/>
        <param name="is_publish_bounding_box" value="$(arg is_publish_bounding_box)"/>

        <param name="score_threshold" value="$(arg score_threshold)"/>
        <param name="iou_threshold" value="$(arg iou_threshold)"/>

        <param name="weights_path" value="$(arg weights_path)"/>
        <param name="config" value="$(arg config)"/>
        <param name="device" value="$(arg device)"/>

        <remap from="/input/image/compressed" to="/hsrb/head_rgbd_sensor/rgb/image_rect_color/compressed"/>
        <remap from="/input/activation" to="/yolov5_ros/input/activation"/>
        <remap from="/output/image/compressed" to="/yolov5_ros/output/image/compressed"/>
        <remap from="/output/bounding_boxes" to="/yolov5_ros/output/bounding_boxes"/>
	<remap from="/output/class_prob" to="/yolov5_ros/output/class_prob"/>
        <remap from="/check_for_objects" to="/yolov5_ros/check_for_objects"/>

    </node>

</launch>
